src/server/mod.rs

```diff
@@ -60,19 +64,6 @@ pub struct Ack;
 #[derive(Debug)]
 pub struct NoResponse;
 
-/// Empty extra parameters to some request or notification.
-#[derive(Debug, Serialize, PartialEq)]
-pub struct NoParams;
```

Comment:
I don't think we can replace `NoParams` with `()`, iirc, the former is serialised as `{}` which is what clients send/expect

Reply:
Per http://www.jsonrpc.org/specification#request_object params can be omitted or be a Structured value (`Array | Object`), so omitting should be legal.
I'll add a test or two for this